project_resource_calendar/models/calendar_event.py

```diff
@@ -371,3 +352,54 @@ def unlink(self):
                     )
                 )
         return super(CalendarEvent, self).unlink()
+
+    def is_hr_resource_double_booked(self, attendee,
+                                     date_start=None, date_end=None):
+        if not date_end and not date_start:
+            date_start = self.date_start
+            date_end = self.date_end
+
+        overlaps_partners = self.env['calendar.event'].search([
+            ('id', '!=', self.id),
```

Comment:
we need to factor this redandant code, we can use a constatnt (OVERLAP_DOMAIN = [
            ('id', '!=', self.id),('start', '<', date_end),
            ('stop', '>', date_start),
            ('state', '!=', 'cancelled'),] and use it each time (we simply add the specific fields to check ( ('partner_ids', 'in', attendee.id))